sweep/sweeper.go

```diff
@@ -750,17 +751,100 @@ func (s *UtxoSweeper) bucketForFeeRate(
 	return 1 + int(feeRate-s.relayFeeRate)/s.cfg.FeeRateBucketSize
 }
 
+// createInputClusters creates a list of input clusters from the set of pending
+// inputs known by the UtxoSweeper. It clusters inputs by
+// 1) Required tx locktime
+// 2) Similar fee rates
+func (s *UtxoSweeper) createInputClusters() []inputCluster {
+	inputs := s.pendingInputs
+
+	// We start by getting the inputs clusters by locktime. Since the
+	// inputs commit to the locktime, they can only be clustered together
+	// if the locktime is equal.
+	lockTimeClusters, inputs := s.clusterByLockTime(inputs)
```

Comment:
Style nit: should rename the return value here to something like `nonLockTimeInputs` to make the partitioning a bit clearer.